adhocracy4/phases/models.py

```diff
@@ -119,15 +114,9 @@ def is_over(self):
     def has_feature(self, feature, model):
         return content[self.type].has_feature(feature, model)
 
-    def is_first_of_project(self):
-        """Test if this is the first phase of the project.
-
-        Attention: _deprecated_ as this only works for one module where the
-        phase order is determined by weight.
-        """
-        warnings.warn(
-            "is_first_of_project is deprecated as it relies "
-            "on the project only having one module",
-            DeprecationWarning
+    def starts_first_of_project(self):
+        """Test if this is the phase that starts first in the project."""
+        phases = self.module.project.phases.order_by(
+            F('start_date').asc(nulls_last=True)
```

Comment:
@Rineee @goapunk It was good, that I added more tests, because doing that I noticed that phases without dates were actually ordered first. I changed that. 
Now there is only the edge-case left when there are only phases without dates. Then a phase without date will be the "first one starting", which might be surprising, but is also not wrong. Especially if there is only one phase it's right. But I think, we can leave it like that.
And then I found another surprising thing, I fixed and got carried away with the tests a bit...

Reply:
@Rineee And now it is really finished (I hope!!!!!!11!!!!! :p) and will filter out all phases from unpublished modules. :)